Casks/o/oracle-jdk.rb

```diff
@@ -1,18 +1,26 @@
 cask "oracle-jdk" do
   arch arm: "aarch64", intel: "x64"
 
-  version "23.0.2"
-  sha256 arm:   "1c63d9d1f19c444845fd2e71c0b16d53b36d627584002fb43979fdeaca3fd33c",
-         intel: "5afa5408ec83310c1407c0d9c65b474f65e56447eece4d1b1b29c36b95ee5a29"
+  version "24"
+  sha256 arm:   "009695152a1f6d967f1f3488fd7f26761f31601ea85e5dfea4d83e4aa2c36289",
+         intel: "fc396bc72ec56659a6889191c871d754ac18e694ce17ba859194ebf7cab71632"
 
   url "https://download.oracle.com/java/#{version.major}/archive/jdk-#{version}_macos-#{arch}_bin.dmg"
   name "Oracle Java Standard Edition Development Kit"
   desc "JDK from Oracle"
   homepage "https://www.oracle.com/java/technologies/downloads/"
 
   livecheck do
-    url "https://www.oracle.com/java/technologies/javase/#{version.major}u-relnotes.html"
+    url "https://www.oracle.com/java/technologies/downloads/"
     regex(/<li>\s*JDK\s*v?(\d+(?:\.\d+)*)/i)
+    strategy :page_match do |page, regex|
+      major = page.scan(%r{href=.*?/javase/(\d+)u-relnotes\.html}i)&.map { |match| match[0].to_i }&.max
+      next if major.blank?
+
+      url = "https://www.oracle.com/java/technologies/javase/#{major}u-relnotes.html"
+      version_page = Homebrew::Livecheck::Strategy.page_content(url)
+      version_page[:content]&.scan(regex)&.map { |match| match[0] }
```

Comment:
```suggestion
      major = page.scan(%r{href=.*?/javase/(\d+)u-relnotes\.html}i)
                  .max_by { |match| Version.new(match[0]) }
                  &.first
      next if major.blank?

      release_page = Homebrew::Livecheck::Strategy.page_content(
        "https://www.oracle.com/java/technologies/javase/#{major}u-relnotes.html",
      )
      next if (release_content = release_page[:content]).blank?

      release_content.scan(regex).map { |match| match[0] }
```

This brings the `strategy` block more in line with existing patterns along with the following:

* Remove the unnecessary safe navigation operator after `String#scan`, as it always returns an array
* Inline the `url` value, as it's not used again
* Use the `next if (release_content = release_page[:content]).blank?` guard pattern, which memoizes the value while checking for existence. This approach saves us from having to use multiple safe navigation operators, which are more ambiguous in this context.